pkg/controller/endpoint/endpoints_controller_test.go

```diff
@@ -343,24 +343,61 @@ func TestCheckLeftoverEndpoints(t *testing.T) {
 	testServer, _ := makeTestServer(t, ns)
 	defer testServer.Close()
 	endpoints := newController(testServer.URL, 0*time.Second)
-	endpoints.endpointsStore.Add(&v1.Endpoints{
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go endpoints.Run(1, stopCh)
+	// cache.WaitForNamedCacheSync has a 100ms poll period, and the endpoints worker has a 10ms period.
+	// To ensure we get all updates, including unexpected ones, we need to wait at least as long as
+	// a single cache sync period and worker period, with some fudge room.
+	time.Sleep(150 * time.Millisecond)
+	endpoints.onEndpointsAdd(&v1.Endpoints{
```

Comment:
this is a dumb test, I want to check how it behaves in the CI and if the approach is correct.
If this is good to go I have to add more scenarios verifying this solve all the referenced bugs and is backwards compatible